stdlib/tests/collections/mmr.rs

```diff
@@ -213,3 +213,140 @@ fn test_mmr_tree_with_one_element() -> Result<(), MerkleError> {
+    let els = hash.as_elements();
+    let size = RpoDigest::new([els[0], els[1], els[2], els[3] + Felt::new(1)]);
+
+    // Set up the VM stack with the MMR hash, and its target address
+    let mut stack = stack_to_ints(&*hash);
+    let mmr_ptr = 1000;
+    stack.insert(0, mmr_ptr);
+
+    // both the advice stack and merkle store start empty (data is available in
+    // the map and pushed to the advice stack by the MASM code)
+    let advice_stack = &[];
+    let store = MerkleStore::new();
+
+    // the advice_map data uses stack order, so every word pair needs to be reversed
+    let map_data: Vec<Felt> = hash_data.as_slice().concat();
+
+    let advice_map: &[([u8; 32], Vec<Felt>)] = &[
+        // The key for the MMR contains the data used during the hashing, which includes the peaks
+        // themselves and the padding data
+        (hash.as_bytes(), map_data),
+        // the key for the size contains a single element, the number_of_leaves of the MMR
+        (size.as_bytes(), vec![forest]),
+    ];
+
+    let source = "
+        use.std::collections::mmr
+        begin exec.mmr::unhash end
+    ";
+    let test = build_test!(source, &stack, advice_stack, store, advice_map.iter().cloned());
+
+    #[rustfmt::skip]
+    let expect_memory = [
+        21, 0, 0, 0, // MMR leaves (only one Felt is used)
+        0, 0, 0, 1,  // first peak
+        0, 0, 0, 2,  // second peak
+        0, 0, 0, 3,  // third peak
+    ];
+    test.expect_stack(&[]);
+    test.expect_stack_and_memory(&[], mmr_ptr, &expect_memory);
+}
+
+#[test]
+fn test_mmr_unhash_invalid_hash() {
+    let number_of_leaves = 0b10101; // 3 peaks, 21 leaves
+    let forest = Felt::new(number_of_leaves);
+
+    // The hash data is not the same as the peaks, it is padded to 16 elements
+    let mut hash_data: [[Felt; 4]; 16] = [
+        // 3 peaks. These hashes are invalid, we can't produce data for any of these peaks (only
+        // for testing)
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::new(1)],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::new(2)],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::new(3)],
+        // Padding, the MMR is padded to a minimum length o 16
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+        [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::ZERO],
+    ];
```

Comment:
initially I didn't have this wall of elements. I added it because it made it easier for me to debug (by seeing were the elements 1,2,3 ended up).

This can be simplified if necessary.